velox/exec/tests/utils/ExpressionBuilder.h

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/parse/Expressions.h"
+
+namespace facebook::velox::expr::builder {
+
+/// Fluent Expression Builder.
+///
+/// This file contains fluent methods that make it convenient to create
+/// (untyped) expression trees. This provides similar functionality to a SQL
+/// parser, without briging dependency on external libraries or bringing leaked
+/// semantics from other systems.
+///
+/// The untyped expressions can then be turned into typed expressions ready for
+/// execution using type binding from `core::Expressions::inferTypes()`.
+///
+/// The API provided is as close to the actual expression trees as possible.
+/// Filters, arithmetics, function calls, literals, aliases, and more are
+/// supported with this API.
+///
+/// For example, to create a field reference (reference to a column), one can:
+///
+/// > using namespace expr::builder;
+/// > ExprPtr e = field("col");
+///
+/// Or use the "_f" literal provided:
+///
+/// > ExprPtr e = "col"_f;
+///
+/// To debug the expression generated, you can simply:
+///
+/// > LOG(INFO) << *e;
+///
+/// Filters (comparisons) can be fluently created using C++ overloaded
+/// operators
+///
+/// > ExprPtr e = "col"_f > 10; // "col > 10"
+/// > ExprPtr e = "col"_f != "bar"; // "col != 'bar'"
+///
+/// Null checking filters:
+///
+/// > ExprPtr e = isNull("col"_f) // "col is null"
+/// > ExprPtr e = !isNull("col"_f) // "col is not null"
+///
+/// Conjuncts and "between":
+///
+/// > ExprPtr e = "a"_f && "b"_f || "c"_f; // "a AND b OR c"
+/// > ExprPtr e = between("a"_f, 0L, 10L); // "a between 0 and 10"
+///
+/// C++ literals are automatically converted into ConstantExpr (expression
+/// literals) when part of an expression. To explicitly create a literal:
+///
+/// > ExprPtr e = literal(10.3);
```

Comment:
Would you add some examples of complex types (arrays, maps and structs)?

Would be nice to clarify how to define literals of various integer types.